### PR TITLE
[sentry-native] Update port to 0.4.2

### PR DESCRIPTION
**Pull request by AenBleidd**
Remove custom fix for msvc 19.27 because of the fix that was introduced by port authors here: https://github.com/getsentry/sentry-native/commit/f1437333d4ea1eca2f8a3664d57efff081b4dad6

Signed-off-by: Vitalii Koshura <lestat.de.lionkur@gmail.com>

### CODE REVIEW

File: ports/sentry-native/portfile.cmake [hunk 1]
```diff
@@ -24,13 +24,6 @@ if (NOT DEFINED SENTRY_BACKEND)
     endif()
 endif()
 
-if (VCPKG_TARGET_IS_WINDOWS)
-    set(SENTRY_NATIVE_C_STANDARD 99)
-else()
-    set(SENTRY_NATIVE_C_STANDARD 11)
-endif()
-
-
```
Comment (ras0219-msft) [inline ports/sentry-native/portfile.cmake:33]: Please also remove the `-DCMAKE_C_STANDARD=...` from the `vcpkg_configure_cmake()` below

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
LGTM once my comment is addressed!

File: ports/sentry-native/portfile.cmake (hunk 1 above)
Comment (AenBleidd) [reply ports/sentry-native/portfile.cmake:33]: Fixed

**Review by JackBoosY — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Thanks for the update!

**Pull request merged**